File: bankApp/src/main/java/entities/Address.java
package entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "addresses")
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false, unique = true, updatable = false)
    private UUID id;

    @Column(name = "country", nullable = false)
    private String country;

    @Column(name = "sector")
    private String sector;

    @Column(name = "city", nullable = false)
    private String city;

    @Column(name = "street")
    private String street;

    @Column(name = "number", nullable = false)
    private String number;

    @Column(name = "zip")
    private String zip;

}

File: frontend/src/classes/Address.ts
export default class Address {
  private _country: string;
  private _sector: string;
  private _city: string;
  private _street: string;
  private _number: string;
  private _zip: string;

  constructor(
    country: string,
    sector: string,
    city: string,
    street: string,
    number: string,
    zip: string,
  ) {
    this._country = country;
    this._sector = sector;
    this._city = city;
    this._street = street;
    this._number = number;
    this._zip = zip;
  }

  public get country(): string {
    return this._country;
  }

  public set country(value: string) {
    this._country = value;
  }

  public get sector(): string {
    return this._sector;
  }

  public set sector(value: string) {
    this._sector = value;
  }

  public get city(): string {
    return this._city;
  }

  public set city(value: string) {
    this._city = value;
  }

  public get street(): string {
    return this._street;
  }

  public set street(value: string) {
    this._street = value;
  }

  public get number(): string {
    return this._number;
  }

  public set number(value: string) {
    this._number = value;
  }

  public get zip(): string {
    return this._zip;
  }

  public set zip(value: string) {
    this._zip = value;
  }
}
